pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -128,6 +128,18 @@ def sync_batchnorm(self) -> bool:
     def sync_batchnorm(self, sync_batchnorm: bool) -> None:
         self._sync_batchnorm = sync_batchnorm
 
+    @property
+    def task_idx(self) -> Optional[int]:
+        rank_zero_deprecation(
+            f'`{self.__class__.__name__}.task_idx` is deprecated in v1.4 and will be removed in v1.6. Use '
+            f'`{self.__class__.__name__}.local_rank` instead.'
+        )
+        return self._task_idx
```

Comment:
Why not just defer to `self.local_rank` here and remove `_task_idx`?

Reply:
I think because of the setter, in which case we don't want to let them set the local rank via task_idx

Reply:
Would technically be backwards incompatible.

So in that case I would just remove the code entirely. This is really old and specific